### PR TITLE
Simplified release process by hosting javadoc on javadoc.io

### PR DESCRIPTION
**Pull request by szczepiq**
1. As we started hosting javadoc on javadoc.io (see #692) we don't need to push javadoc to gh-pages any more. This makes the releases simpler and faster.
2. I sneaked in test fixes after Brice caught a problem in my previous change :). Thanks!!!

Tested by dry-running the release locally:
```
Dry-running the release
Performing 7 release steps
Step 1: ensure good chunk of recent commits is pulled for release notes automation
--- Executing: git pull --depth 1000 origin release/2.x
From https://github.com/mockito/mockito
 * branch            release/2.x -> FETCH_HEAD
Already up-to-date.
--- Completed!
Step 2: configure generic git user
Step 3: update release notes
Building new release notes based on /Users/sfaber/mockito/src/doc/release-notes/official.md
Building notes for revisions: v2.2.7 -> HEAD
Successfully updated release notes!
Step 4: commit release notes
--- Executing: git add /Users/sfaber/mockito/src/doc/release-notes/official.md
--- Completed!
--- Executing: git commit -m "Update release notes by Travis CI build null [ci skip]" /Users/sfaber/mockito/src/doc/release-notes/official.md
[issue-692 1b12082] "Update release notes by Travis CI build null [ci skip]"
 1 file changed, 7 insertions(+)
--- Completed!
Step 5: create new version tag
--- Executing: git tag -a v2.2.8 -m Create tag v2.2.8 by Travis CI build null [ci skip]
--- Completed!
Step 6: commit incremented version on release/2.x
--- Executing: git commit -m Increment version '2.2.8' -> '2.2.9' by Travis CI build null [ci skip] version.properties
[issue-692 df2d903] Increment version '2.2.8' -> '2.2.9' by Travis CI build null [ci skip]
 1 file changed, 1 insertion(+), 1 deletion(-)
--- Completed!
Step 7: push changes to all involved branches
--- Executing: git push <masked> release/2.x v2.2.8 -q --dry-run
--- Completed!
No cleanup found for step 'push changes to all involved branches'
No cleanup found for step 'commit incremented version on release/2.x'
No cleanup found for step 'create new version tag'
No cleanup found for step 'commit release notes'
No cleanup found for step 'update release notes'
Cleaning up after step 'configure generic git user'
No cleanup found for step 'ensure good chunk of recent commits is pulled for release notes automation'
:rollbackRelease
Release failed. Rolling back 7 release steps.
Attempting to roll back step 7 (push changes to all involved branches)
No rollback or cleanup operation found for step 'push changes to all involved branches'
Attempting to roll back step 6 (commit incremented version on release/2.x)
--- Executing: git reset --hard HEAD~1
HEAD is now at 1b12082 "Update release notes by Travis CI build null [ci skip]"
--- Completed!
Attempting to roll back step 5 (create new version tag)
--- Executing: git tag -d v2.2.8
Deleted tag 'v2.2.8' (was f05a82e)
--- Completed!
Attempting to roll back step 4 (commit release notes)
--- Executing: git reset --hard HEAD~1
HEAD is now at d2e1b4b Made release easier to test
--- Completed!
Attempting to roll back step 3 (update release notes)
No rollback or cleanup operation found for step 'update release notes'
Attempting to roll back step 2 (configure generic git user)
Attempting to roll back step 1 (ensure good chunk of recent commits is pulled for release notes automation)
No rollback or cleanup operation found for step 'ensure good chunk of recent commits is pulled for release notes automation'
```

### CODE REVIEW

File: gradle/release.gradle [hunk 1]
```diff
@@ -159,21 +157,19 @@ releaseSteps {
              - see 'TROUBLESHOOTING' section at the top of this file
     */
 
-    step("start operating on $System.env.TRAVIS_BRANCH") { run "git", "checkout",  System.env.TRAVIS_BRANCH }
-
     step("update release notes") { project.notes.updateReleaseNotes() }
 
     step("commit release notes") { commitReleaseNotes(buildInfo) }
-            .rollback { run "git", "reset", "--hard", "HEAD^" }
+            .rollback { run "git", "reset", "--hard", "HEAD~1" }
```
Comment (szpak) [inline gradle/release.gradle]: `HEAD~` would be also fine.

File: src/test/java/org/mockitousage/debugging/InvocationsPrinterTest.java [hunk 2]
```diff
@@ -9,36 +8,35 @@
...
-import static org.junit.Assert.assertEquals;
 import static org.mockito.Mockito.when;
 
 public class InvocationsPrinterTest extends TestBase {
 
     @Mock IMethods mock;
 
     @Test public void no_invocations() {
-        Assert.assertEquals("No interactions and stubbings found for mock: mock", new InvocationsPrinter().printInvocations(mock));
+        assertThat(new InvocationsPrinter().printInvocations(mock)).isEqualTo("No interactions and stubbings found for mock: mock");
```
Comment (szpak) [inline src/test/java/org/mockitousage/debugging/InvocationsPrinterTest.java]: Those changes don't seem to be closely related to javadoc.io.

File: gradle/release.gradle (hunk 1 above)
Comment (bric3) [inline gradle/release.gradle]: imho I prefer by little bit without preference either `HEAD^` or `HEAD~1` which I find easy to understand over `HEAD~`

File: src/test/java/org/mockitousage/debugging/InvocationsPrinterTest.java (hunk 2 above)
Comment (szczepiq) [inline src/test/java/org/mockitousage/debugging/InvocationsPrinterTest.java]: Correct, they are completely unrelated :)

**Pull request merged**